File: engenharia_de_software_III/atividade_3/Q1/RepositorioDePosts.java
package atividade_3.Q1;

import java.io.FileWriter;
import java.io.IOException;

public class RepositorioDePosts {
    public void saveToFile(Post post) {
        try (FileWriter writer = new FileWriter(post.getFilePath())) {
            writer.write("ID: " + post.getId() + "\n");
            writer.write("Texto: " + post.getTexto() + "\n");
            writer.write("Autor: " + post.getAutor().getNome() + "\n");
            writer.write("Data: " + post.getData() + "\n");
            writer.write("Quantidade de Likes: " + post.getFilePath() + "\n");
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: engenharia_de_software_III/atividade_3/Q1/RepositorioDePosts.ts
import * as fs from 'fs';
import { Post } from './Post';

class RepositorioDePosts {
    saveToFile(post: Post): void {
        try {
            const writer = fs.createWriteStream(post.getFilePath());
            writer.write(`ID: ${post.getId()}\n`);
            writer.write(`Texto: ${post.getTexto()}\n`);
            writer.write(`Autor: ${post.getAutor().getNome()}\n`);
            writer.write(`Data: ${post.getData()}\n`);
            writer.write(`Quantidade de Likes: ${post.getQuantidadeDeLikes()}\n`);
            writer.close();
        } catch (error) {
            console.error(error);
        }
    }
}

export default RepositorioDePosts;
